StyleCop.Analyzers/StyleCop.Analyzers.Test.CSharp8/SpacingRules/SA1011CSharp8UnitTests.cs

```diff
@@ -52,6 +52,26 @@ public class TestClass
         }
     }
 }
+";
+
+            await VerifyCSharpDiagnosticAsync(LanguageVersion.CSharp8, testCode, DiagnosticResult.EmptyDiagnosticResults, CancellationToken.None).ConfigureAwait(false);
+        }
+
+        [Fact]
+        [WorkItem(3052, "https://github.com/DotNetAnalyzers/StyleCopAnalyzers/issues/3052")]
+        public async Task TestClosingSquareBracketFollowedByExclamationAsync()
+        {
+            var testCode = @"namespace TestNamespace
+{
+    public class TestClass
+    {
+        public void TestMethod(object?[] arguments)
+        {
+            object o = arguments[0]!;
+            string s = arguments[0]!.ToString();
```

Comment:
:bulb: can you add a test showing that a space here would get removed? A space after the `!` would likely be reported on the `.` or `;` token since those have dedicated diagnostic IDs, but a space before might be part of this analyzer. The tests should ideally be added to all the impacted diagnostics.

Reply:
Yes, I updated PR.